### DISCUSSION TITLE
Issues installing netbox - (code=exited, status=203/EXEC)

### DISCUSSION THREAD

**Discussion by abanta1** [netbox-community/netbox#8606] [Q&A] [answered]
I'm having issues installing netbox. I get as far as the Gunicorn steps and receive an error when attempting to start the netbox service (code=exited, status=203/EXEC). I'm having the same issue as this post: https://groups.google.com/g/netbox-discuss/c/yh6Hb69K-oo

I've attached the journalctl and systemctl status outputs

Any advice?

[journalctl.txt](https://github.com/netbox-community/netbox/files/8043067/journalctl.txt)

**Comment by candlerb** [comment]
Code 203 from systemd simply means "unable to start the process".  Maybe it doesn't exist at the path you gave, or maybe permissions are set wrongly.

I suggest you test this by running the "ExecStart" line directly from the command line, making sure you do this as the "netbox" user (assuming your systemd unit file has `User=netbox`).  Like this:

```
sudo -u netbox /opt/netbox/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/netbox/netbox --config /opt/netbox/gunicorn.py netbox.wsgi
```

What response do you get from that?

**Comment by abanta1** [comment]
[abanta@netbox-p001  ~]$ sudo -u netbox /opt/netbox/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/netbox/netbox --config /opt/netbox/gunicorn.py netbox.wsgi
[sudo] password for abanta:

Error: [Errno 13] Permission denied: '/home/abanta'
[abanta@netbox-p001  ~]$

however, if I run it as myself:

[abanta@netbox-p001 ~]$ /opt/netbox/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/netbox/netbox --config /opt/netbox/gunicorn.py netbox.wsgi
[2022-02-11 11:48:52 -0700] [636796] [INFO] Starting gunicorn 20.1.0
[2022-02-11 11:48:52 -0700] [636796] [INFO] Listening at: http://127.0.0.1:8001 (636796)
[2022-02-11 11:48:52 -0700] [636796] [INFO] Using worker: gthread
[2022-02-11 11:48:52 -0700] [636797] [INFO] Booting worker with pid: 636797
[2022-02-11 11:48:52 -0700] [636798] [INFO] Booting worker with pid: 636798
[2022-02-11 11:48:53 -0700] [636799] [INFO] Booting worker with pid: 636799
[2022-02-11 11:48:53 -0700] [636800] [INFO] Booting worker with pid: 636800
[2022-02-11 11:48:53 -0700] [636801] [INFO] Booting worker with pid: 636801
^C[2022-02-11 11:48:59 -0700] [636796] [INFO] Handling signal: int
[2022-02-11 18:48:59 +0000] [636799] [INFO] Worker exiting (pid: 636799)
[2022-02-11 18:48:59 +0000] [636797] [INFO] Worker exiting (pid: 636797)
[2022-02-11 18:48:59 +0000] [636800] [INFO] Worker exiting (pid: 636800)
[2022-02-11 18:48:59 +0000] [636801] [INFO] Worker exiting (pid: 636801)
[2022-02-11 18:48:59 +0000] [636798] [INFO] Worker exiting (pid: 636798)
[2022-02-11 11:48:59 -0700] [636796] [INFO] Shutting down: Master
[abanta@netbox-p001 ~]$

**Comment by baldgeek** [comment]
I believe this is an error from where you are running the command. Try 
doing 'cd /tmp' before doing sudo

**Reply by abanta1** [reply]
changing to /tmp did allow it to work:

[abanta@netbox-p001 tmp]$ sudo -u netbox /opt/netbox/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/netbox/netbox --config /opt/netbox/gunicorn.py netbox.wsgi
[2022-02-11 13:04:42 -0700] [640582] [INFO] Starting gunicorn 20.1.0
[2022-02-11 13:04:42 -0700] [640582] [INFO] Listening at: http://127.0.0.1:8001/ (640582)
[2022-02-11 13:04:42 -0700] [640582] [INFO] Using worker: gthread
[2022-02-11 13:04:42 -0700] [640583] [INFO] Booting worker with pid: 640583
[2022-02-11 13:04:42 -0700] [640584] [INFO] Booting worker with pid: 640584
[2022-02-11 13:04:42 -0700] [640585] [INFO] Booting worker with pid: 640585
[2022-02-11 13:04:42 -0700] [640586] [INFO] Booting worker with pid: 640586
[2022-02-11 13:04:42 -0700] [640587] [INFO] Booting worker with pid: 640587
^C[2022-02-11 13:04:48 -0700] [640582] [INFO] Handling signal: int
[2022-02-11 20:04:48 +0000] [640585] [INFO] Worker exiting (pid: 640585)
[2022-02-11 20:04:48 +0000] [640583] [INFO] Worker exiting (pid: 640583)
[2022-02-11 20:04:48 +0000] [640587] [INFO] Worker exiting (pid: 640587)
[2022-02-11 20:04:48 +0000] [640584] [INFO] Worker exiting (pid: 640584)
[2022-02-11 20:04:48 +0000] [640586] [INFO] Worker exiting (pid: 640586)
[2022-02-11 13:04:49 -0700] [640582] [INFO] Shutting down: Master
[abanta@netbox-p001 tmp]$

**Comment by candlerb** [comment]
Can you show your entire netbox.service?  Another possibility is that the netbox user doesn't have access to the WorkingDirectory or home directory set for that user.

**Reply by abanta1** [reply]
netbox.service attached
[netbox.service.txt](https://github.com/netbox-community/netbox/files/8054165/netbox.service.txt)

[abanta@netbox-p001 ~]$ ls -ld /opt/netbox
drwxr-xr-x. 9 root root 4096 Feb  2 16:52 /opt/netbox
[abanta@netbox-p001 ~]$ cat /etc/passwd | grep netbox
netbox: x:989:985::/home/netbox:/bin/bash
[abanta@netbox-p001 ~]$ ls -ld /home/netbox
drwx------. 2 netbox netbox 83 Feb  2 16:58 /home/netbox
[abanta@netbox-p001 ~]$

**Comment by candlerb** [comment]
> $ cat /etc/passwd | grep netbox
> netbox: x:989:985::/home/netbox:/bin/bash

There appears to be a spurious space there. I don't know if that will cause a problem.

Check these:

```
grep netbox /etc/group
ls -l /opt/netbox/venv/bin
```

I'm pretty sure this is *some* sort of permissions problem.

**Reply by abanta1** [reply]
Yes, the GitHub interpreter changed it to an image so I added a space when posting.

```
[abanta@netbox-p001 ~]$ grep netbox /etc/group
netbox:x:985:
[abanta@netbox-p001 ~]$ ls -l /opt/netbox/venv/bin/
total 104
-rw-r--r--. 1 root root 1983 Feb  2 16:46 activate
-rw-r--r--. 1 root root  909 Feb  2 16:46 activate.csh
-rw-r--r--. 1 root root 2051 Feb  2 16:46 activate.fish
-rw-r--r--. 1 root root 9033 Feb  2 16:46 Activate.ps1
-rwxr-xr-x. 1 root root  278 Feb  2 16:46 django-admin
-rwxr-xr-x. 1 root root  668 Feb  2 16:46 django-admin.py
-rwxr-xr-x. 1 root root  224 Feb  2 16:46 ghp-import
-rwxr-xr-x. 1 root root  232 Feb  2 16:46 gunicorn
-rwxr-xr-x. 1 root root  228 Feb  2 16:46 jsonschema
-rwxr-xr-x. 1 root root  229 Feb  2 16:46 markdown_py
-rwxr-xr-x. 1 root root  227 Feb  2 16:46 mkdocs
-rwxr-xr-x. 1 root root  225 Feb  2 16:46 netaddr
-rwxr-xr-x. 1 root root  259 Feb  2 16:46 normalizer
-rwxr-xr-x. 1 root root  236 Feb  2 16:46 pip
-rwxr-xr-x. 1 root root  236 Feb  2 16:46 pip3
-rwxr-xr-x. 1 root root  236 Feb  2 16:46 pip3.10
drwxr-xr-x. 2 root root 4096 Feb  2 16:46 __pycache__
-rwxr-xr-x. 1 root root  230 Feb  2 16:46 pygmentize
lrwxrwxrwx. 1 root root   10 Feb  2 16:46 python -> python3.10
lrwxrwxrwx. 1 root root   10 Feb  2 16:46 python3 -> python3.10
lrwxrwxrwx. 1 root root   19 Feb  2 16:46 python3.10 -> /usr/bin/python3.10
-rwxr-xr-x. 1 root root  220 Feb  2 16:46 rq
-rwxr-xr-x. 1 root root  220 Feb  2 16:46 rqinfo
-rwxr-xr-x. 1 root root  224 Feb  2 16:46 rqworker
-rwxr-xr-x. 1 root root  231 Feb  2 16:46 sqlformat
-rwxr-xr-x. 1 root root  232 Feb  2 16:46 watchmedo
-rwxr-xr-x. 1 root root  223 Feb  2 16:46 wheel
[abanta@netbox-p001 ~]$
```

**Comment by markkuleinio** [comment]
Is SELinux in use?

**Reply by abanta1** [reply]
Not that I'm aware of. Is there a command I can run to verify?

**Reply by abanta1** [reply]
Disregard, found it and it appears it is in use:

```
[abanta@netbox-p001 ~]$ sestatus
SELinux status:                 enabled
SELinuxfs mount:                /sys/fs/selinux
SELinux root directory:         /etc/selinux
Loaded policy name:             targeted
Current mode:                   enforcing
Mode from config file:          enforcing
Policy MLS status:              enabled
Policy deny_unknown status:     allowed
Memory protection checking:     actual (secure)
Max kernel policy version:      33
[abanta@netbox-p001 ~]$
```

**Comment by abanta1** [comment, marked as answer]
Figured it out - 

Had to run

```
[abanta@netbox-p001 ~]$ journalctl -xe
Feb 16 11:27:48 netbox-p001 setroubleshoot[2618]: SELinux is preventing /usr/lib/systemd/systemd from execute access on the file /opt/netbox/venv/bin/gunicorn. For complete SELinux messages run: sealert -l 58450263-d342-4941-98a3-47bd8d91dc73
Feb 16 11:27:48 netbox-p001 setroubleshoot[2618]: SELinux is preventing /usr/lib/systemd/systemd from execute access on the file /opt/netbox/venv/bin/gunicorn.
```
then
```
[abanta@netbox-p001 ~]$ sealert -l 58450263-d342-4941-98a3-47bd8d91dc73
SELinux is preventing /usr/lib/systemd/systemd from execute access on the file /opt/netbox/venv/bin/gunicorn.

*****  Plugin restorecon (71.4 confidence) suggests   ************************

If you want to fix the label.
/opt/netbox/venv/bin/gunicorn default label should be bin_t.
Then you can run restorecon. The access attempt may have been stopped due to insufficient permissions to access a parent directory in which case try to change the following command accordingly.
Do
# /sbin/restorecon -v /opt/netbox/venv/bin/gunicorn
```
and finally
```
[abanta@netbox-p001 ~]$ /sbin/restorecon -v /opt/netbox/venv/bin/gunicorn
[sudo] password for abanta:
Relabeled /opt/netbox/venv/bin/gunicorn from unconfined_u:object_r:unlabeled_t:s0 to unconfined_u:object_r:bin_t:s0
```

Then stopped and started the service and its working now

**Reply by chicks-net** [reply]
Thanks for documenting how to get around these SELinux issues.
